core/baseline-core-v1.0-psd01.md

```diff
@@ -2786,113 +2795,722 @@ The document defines storage capability requirements in the following areas:
+3. Verify that the storage nodes can communicate securely using the established relationship.
+4. Attempt to access the data stored on one storage node from the other without using the established relationship and verify that it is not possible.
+5. Repeat steps 2-4 for different pairs of storage nodes in the network.
+
+Expected Results:
+1. Each storage node has a unique identity and a corresponding public key.
+2. The pairwise key/identity relationships between storage nodes are established.
+3. The storage nodes can communicate securely using the established relationships.
+4. It is not possible to access the data stored on one storage node from the other without using the established relationship.
+
+The requirement is met, if all expected results are met. The requirement is not me, if any of the results are not meeting test expectations.
+
 #### **[R282]**	
 BPI Storage MUST be compatible with commonly used external authentication services. 
 
 *Non-normative examples of such authentication technologies are OAUTH [[OAuth-2.0](#oauth-20)] , SAML [[SAML](#saml)] , OIDC [[OIDC](#oidc)], AD/LDAP [[ActiveDirectory](#activedirectory)].*
 
+[[R282]](#r282) Testability: 
+
+Preconditions: 
+
+* A typical BPI Storage e.g. the open-source [PostgreSQL](https://www.postgresql.org/) data base deployed in a BPI test environment with a BPI client and a BPI server
+* An authemtication service to be used with the BPI Storage such as [[OAuth-2.0](#oauth-20)] between client and server.
+
+Test steps:
+
+1. Create a BPI Subject within the BPI with a public-private cryptographic key pair based on one of the cryptographic algorithms supported by the authentication service chosen for the BPI Subject
+2. Create an authentication policy for the BPI Storage supported by each of the authemtication services chosen for the test  
+2. Attempt to authenticate the BPI Subject with the BPI Storage using each of the identified external authentication services for the test.
+3. Verify that the authentication of the BPI Subject is successful and the BPI Subject is granted the appropriate access to the BPI Storage as per the test authentication policy.
+4. Repeat the above steps for each supported external authentication service.
+
+Test pass criteria:
+
+The test will pass if one or more of the external authentication services successfully authenticate a BPI Subject to the BPI Storage and the BPI Subject is granted the type of access to the BPI Storage as defined in the test policy. 
+
+Note that the test policy could be any combination of read and write access to the BPI Storage.
+
 #### **[R283]**	
 BPI Storage MUST support roles & access management.
 
+[[R283]](#r283) Testability: 
+
+Preconditions: 
+
+* Identify the roles that need to be supported by the BPI Storage, based on the functional requirements and design specifications such as "Administrators" with full read and write permissions or "ReadUser" with only read permissions.
```

Comment:
Given that there are some examples, but no clear explanation of the minimum requirements to test, this seems to demonstrate that the requirement is inadequately expressed.

One approach would be to require a minimum set of defined roles, another would be to point to a defined mechanism such as RBAC (or one of its friends), and then test that in detail.

Reply:
rewrote test such that examples could be removed without suffering logical specificity and clarity (hopefully)